src/stream/src/executor/join/row.rs

```diff
@@ -61,6 +68,44 @@ fn build_degree_row(order_key: impl Row, degree: DegreeType) -> impl Row {
     order_key.chain(row::once(Some(ScalarImpl::Int64(degree as i64))))
 }
 
+#[derive(Clone, Debug)]
+pub enum CachedJoinRow {
+    Encoded(EncodedJoinRow),
+    Unencoded(JoinRow<OwnedRow>),
```

Comment:
Just realized one possible further improvement is to make the clone of `JoinRow<OwnedRow>` a shallow copy instead of a deep copy, as `OwnedRow` is `Box<[Datum]>`. 

Is `Arc<Some_Lock<>>` the way to go?

Reply:
There's no need for `SomeLock` as row is designed to be immutable.

However, I'm still dubious if it's a good idea to switch `Box` into `Arc` for `OwnedRow`. In most cases we should avoid cloning a row as much as possible, instead of optimizing the cloning, considering that it introduces atomic operations every time a row is created or dropped.

For this specific case, instead of changing the internal of `OwnedRow`, I think you can try directly using `Arc<[Datum]>` here (by implementing `Row` trait for it). `Row` does not have to be an `OwnedRow`. 😄